When running `yarn install && yarn workspace @automattic/site-admin run build` on this branch I get the following error suggesting we need `@types/wordpress__commands` which doesn't exist. Can you reproduce? 

```
src/components/site-hub/index.tsx:4:40 - error TS7016: Could not find a declaration file for module '@wordpress/commands'. '/Users/psealock/Workspace/calypso/node_modules/@wordpress/commands/build/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/wordpress__commands` if it exists or add a new declaration (.d.ts) file containing `declare module '@wordpress/commands';`
```

--- a/packages/site-admin/package.json
+++ b/packages/site-admin/package.json
@@ -57,6 +57,7 @@
 	},
 	"dependencies": {
 		"@wordpress/base-styles": "^5.20.0",
+		"@wordpress/commands": "^1.19.0",
 		"@wordpress/components": "^29.6.0",
 		"@wordpress/compose": "^7.20.0",
 		"@wordpress/core-data": "^7.20.0",

--- a/packages/site-admin/src/components/site-hub/stories/index.stories.tsx
+++ b/packages/site-admin/src/components/site-hub/stories/index.stories.tsx
@@ -1,6 +1,8 @@
 /**
  * External dependencies
  */
+import { action } from '@storybook/addon-actions';
+import { useEffect, useRef } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 /**
  * Internal dependencies
@@ -32,6 +34,28 @@ export const Default: Story = {
 					isTransparent
 					navigationBackLabel={ __( 'Go to the Dashboard', 'a8c-site-admin' ) }
 					navigationBackLink="/"
+				/>
+			</RouterProvider>
+		);
+	},
+};
+
+export const FocusButtonOnMount: Story = {
+	render: function Template() {
+		const buttonRef = useRef< HTMLButtonElement >( null );
+
+		useEffect( () => {
+			buttonRef.current?.focus();
+			action( 'Focus button' )( 'Focus on button' );
+		}, [] );
+
+		return (
+			<RouterProvider routes={ [] } pathArg="page">
+				<SiteHub
+					isTransparent
+					navigationBackLabel={ __( 'Go to the Dashboard', 'a8c-site-admin' ) }
+					navigationBackLink="/"
+					ref={ buttonRef }
 				/>
 			</RouterProvider>
 		);

--- a/packages/site-admin/src/components/site-hub/style.scss
+++ b/packages/site-admin/src/components/site-hub/style.scss
@@ -93,3 +93,54 @@
 		}
 	}
 }
+
+.site-admin-layout__view-mode-toggle.components-button {
+	view-transition-name: toggle;
+	position: relative;
+	color: $white;
+	height: $header-height;
+	width: $header-height;
+	overflow: hidden;
+	padding: 0;
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	background: $gray-900;
+	border-radius: 0;
+
+	&:hover,
+	&:active {
+		color: $white;
+	}
+
+	&:focus-visible,
+	&:focus {
+		box-shadow: 0 0 0 3px #1e1e1e, 0 0 0 6px var(--wp-admin-theme-color);
+		outline: 4px solid #0000;
+		outline-offset: 4px;
+	}
+
+	&::before {
+		content: "";
+		display: block;
+		position: absolute;
+		top: 9px;
+		right: 9px;
+		bottom: 9px;
+		left: 9px;
+		border-radius: $radius-medium;
+		box-shadow: none;
+
+		@media not (prefers-reduced-motion) {
+			transition: box-shadow 0.1s ease;
+		}
+	}
+
+	.site-admin-layout__view-mode-toggle-icon {
+		display: flex;
+		height: $header-height;
+		width: $header-height;
+		justify-content: center;
+		align-items: center;
+	}
+}

